aidial_adapter_openai/exception_handlers.py

```diff
@@ -1,56 +1,81 @@
-import pydantic
-from aidial_sdk._errors import pydantic_validation_exception_handler
 from aidial_sdk.exceptions import HTTPException as DialException
+from aidial_sdk.exceptions import InternalServerError
 from fastapi import Request
 from fastapi.responses import Response
-from openai import APIConnectionError, APIError, APIStatusError, APITimeoutError
+from openai import (
+    APIConnectionError,
+    APIError,
+    APIStatusError,
+    APITimeoutError,
+    OpenAIError,
+)
 
+from aidial_adapter_openai.utils.adapter_exception import (
+    AdapterException,
+    parse_adapter_exception,
+)
 
-def openai_exception_handler(request: Request, e: DialException):
-    if isinstance(e, APIStatusError):
-        r = e.response
-        headers = r.headers
 
-        # Avoid encoding the error message when the original response was encoded.
-        if "Content-Encoding" in headers:
-            del headers["Content-Encoding"]
+def to_adapter_exception(exc: Exception) -> AdapterException:
+    if isinstance(exc, APIStatusError):
+        # Non-streaming errors reported by `openai` library via this exception
+        r = exc.response
+        httpx_headers = r.headers
 
-        return Response(
-            content=r.content,
+        # httpx library (used by openai) automatically sets
+        # "Accept-Encoding:gzip,deflate" header in requests to the upstream.
+        # Therefore, we may receive from the upstream gzip-encoded
+        # response along with "Content-Encoding:gzip" header.
+        # We either need to encode the response, or
```

Comment:
Minor: we either need to **decode** the response, or remove "Content-Encoding" header

Reply:
No, it's correct. 

We are sending the **decoded** response along with "Content-Encoding:gzip," meaning that the content of the response and the header contradict each other. 

To make them consistent, we need to _EITHER_ **encode** the response using gzip _OR_ remove the "Content-Encoding" header.